llvm/test/CodeGen/NVPTX/nvvm-reflect-options.ll

```diff
@@ -0,0 +1,26 @@
+; Verify that when passing in command-line options to NVVMReflect, that reflect calls are replaced with
```

Comment:
The test is functionally fine, but it also makes me stop and think "what exactly are we doing here and why?".

Two points:
- generating test source for such a small test adds as much complexity as would be adding a copy of the file with additional module options. 
- we're checking for both reflected values encoded into one return value. I would keep them in separate functions, checked individually.
- we're only checking ability to override the values already known to the reflect pass. We need a test case that reflects a non-standard value, and verify what happens when it's passed via command line and when it's not.
- automating the checks would be beneficial as it's sometimes useful to see more of the generated IR than just a single instruction.